timezone/src/modules/Timezone.rb

```diff
@@ -984,6 +986,23 @@ def Summary
       HTML.List(ret)
     end
 
+
+    # Checks whether the system has Windows installed on a primary partition with NTFS
+    def system_has_windows?
+      storage = Y2Storage::StorageManager.instance
+      Storage::Ntfs::all(storage.probed).each do |ntfs|
+        ntfs.blk_devices.each do |blk_device|
+          if Storage::partition?(blk_device)
+            if Storage::to_partition(blk_device).type == Storage::PartitionType_PRIMARY
+              return true if ntfs.detect_content_info.windows?
+            end
+          end
+        end
```

Comment:
@aschnell if you do not give up, I will not give up on C++11! :-)
@ancorgs This is much clearer. In part because it removes the counterintuitive move from a fs to a block device, but also because it expresses queries and names the results. May I also suggest

```rb
ntfs_fss = prim_parts.filesystems.with(type: Storage::FsType_NTFS)
ntfs_fss.any? { |fs| fs.detect_content_info.windows? }
```